code/game/objects/objs.dm

```diff
@@ -403,3 +403,25 @@
 // Stub, used by /item and /structure
 /obj/proc/refresh_color()
 	return
+
+// Used by HE pipes and forging bars/billets. Defaults are for HE pipes.
+/obj/proc/animate_heat_glow(icon_temperature, scale_sub = 500, scale_div = 1500, scale_max = 2000, skip_filter = FALSE)
+
+	var/scale = max((icon_temperature - scale_sub) / scale_div, 0)
+	var/h_r = heat2color_r(icon_temperature)
+	var/h_g = heat2color_g(icon_temperature)
+	var/h_b = heat2color_b(icon_temperature)
+
+	if(icon_temperature < scale_max)
+		h_r = 64 + (h_r - 64)*scale
+		h_g = 64 + (h_g - 64)*scale
+		h_b = 64 + (h_b - 64)*scale
+
+	var/scale_color = rgb(h_r, h_g, h_b)
+	var/list/animate_targets = get_above_oo() + src
+	for (var/thing in animate_targets)
+		var/atom/movable/AM = thing
+		animate(AM, color = scale_color, time = 2 SECONDS, easing = SINE_EASING)
```

Comment:
this is why they're all grey when cold, because `heat2color_foo` gives grey at lower temperatures. need to factor in base color somewhere. maybe lerp between `get_color()` and scale_color?
that or you could maybe make the `heat2color_foo` procs defined on /obj instead of being global procs, and they could use `src.get_color()` as a basis.

Reply:
```dm
		var/list/base_color = rgb2num(get_color())
		h_r = base_color[1] + (h_r - base_color[1])*scale
		h_g = base_color[2] + (h_g - base_color[2])*scale
		h_b = base_color[3] + (h_b - base_color[3])*scale
```